src/goto-instrument/README.md

```diff
@@ -26,3 +29,154 @@ developers are advised to copy the directory, remove all files apart
 from `main.*`, `parseoptions.*` and the `Makefile` and use these as the
 skeleton of their application. The `doit()` method in `parseoptions.cpp`
 is the preferred location for the top level control for the program.
+
+Usage Examples
+--------------
+
+### Main usage ###
+
+For most of the transformations, `goto-instrument` takes 3 arguments: 
+the transformation as an optional flag, then a goto-binary that it uses
+as a source for the transformation, and then *another goto binary* that
+it uses to write the result of the transformation. This is important
+because many times, if you don't supply a second filename for the 
+goto-binary to be written to, you get the equivalent of the `--help`
+option output, with not indication of what has gone wrong. For more specific
+examples, take a look at the demonstrations below:
+
+### Call Graph ###
+
+You can see the call graph of a particular goto-binary by issuing `goto-instrument --call-graph <goto_binary>`. This gives a result similar to this:
```

Comment:
This is quite a surprising _first_ example, because it's one of those that does _not_ require three parameters.